numpy/ma/core.py

```diff
@@ -1613,7 +1613,11 @@ def make_mask(m, copy=False, shrink=True, dtype=MaskType):
             else:
                 result = m
         else:
-            result = np.array(m, dtype=dtype, copy=copy)
+            # existence of fields implies True
+            if m.dtype.fields and np.dtype(dtype) == np.bool_:
+                result = np.ones(m.shape, dtype=dtype)
```

Comment:
This doesn't seem right to me. I'm assuming the aim here is to convert a per-element mask to a per-field mask? This is converting a per-element mask to a "everything is masked" mask